### PR TITLE
Tdr 3248 validate non directory ids submitted

### PR DESCRIPTION
**Pull request by TomJKing**

### CODE REVIEW

File: src/main/scala/uk/gov/nationalarchives/tdr/api/metadatainputvalidation/MetadataInputTags.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+case class ValidateMetadataInput[T](argument: Argument[T]) extends MetadataInputTag {
+
+  override def validateAsync(ctx: Context[ConsignmentApiContext, _])(implicit executionContext: ExecutionContext): Future[BeforeFieldResult[ConsignmentApiContext, Unit]] = {
+    val arg: T = ctx.arg[T](argument.name)
+
+    val inputFileIds: Seq[UUID] = arg match {
+      case input: UpdateBulkFileMetadataInput => input.fileIds
+    }
+    val inputConsignmentId: UUID = arg match {
+      case input: UpdateBulkFileMetadataInput => input.consignmentId
```
Comment (vimleshtna) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/metadatainputvalidation/MetadataInputTags.scala:27]: We can get the from the `inputFileIds` and `inputConsignmentId` from a single matcher 

```
    val (inputFileIds, inputConsignmentId) = arg match {
      case input: UpdateBulkFileMetadataInput => (input.fileIds, input.consignmentId)
    }
```

File: src/main/scala/uk/gov/nationalarchives/tdr/api/metadatainputvalidation/MetadataInputTags.scala [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+      }
+    }
+  }
+
+  private def inputErrors(fileFields: Seq[FileDetails], inputIds: Seq[UUID], inputConsignmentId: UUID): Boolean = {
+    val existingIds = fileFields.map(_.fileId).toSet
+    val consignmentIds = fileFields.map(_.consignmentId).toSet
+    val consignmentIdMismatch = consignmentIds.size != 1 || !consignmentIds.contains(inputConsignmentId)
+    inputIds.toSet.size > existingIds.size || fileFields.exists(_.fileType.contains(NodeType.directoryTypeIdentifier)) || consignmentIdMismatch
+  }
```
Comment (vimleshtna) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/metadatainputvalidation/MetadataInputTags.scala:54]: Looks fine, just to make it more readable:
  ```
private def inputErrors(fileFields: Seq[FileDetails], inputFileIds: Seq[UUID], inputConsignmentId: UUID): Boolean = {
    val existingFileIds = fileFields.map(_.fileId).toSet
    val consignmentIds = fileFields.map(_.consignmentId).toSet
    
    val consignmentIdMismatch = consignmentIds.size != 1 || !consignmentIds.contains(inputConsignmentId)
    lazy val fileIdsMismatch = inputFileIds.toSet.size > existingFileIds.size
    lazy val containsDirectory = fileFields.exists(_.fileType.contains(NodeType.directoryTypeIdentifier))

    consignmentIdMismatch || fileIdsMismatch || containsDirectory
  }
```

File: src/main/scala/uk/gov/nationalarchives/tdr/api/metadatainputvalidation/MetadataInputValidator.scala [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package uk.gov.nationalarchives.tdr.api.metadatainputvalidation
```
Comment (vimleshtna) [inline src/main/scala/uk/gov/nationalarchives/tdr/api/metadatainputvalidation/MetadataInputValidator.scala:1]: Can we rename the package to `uk.gov.nationalarchives.tdr.api.validator` and move all the files related to validation to the `validator` package?

AuthorisationTags.scala
ValidationAuthoriser.scala
ConsignmentStateTags.scala
ConsignmentStateValidator.scala
MetadataInputTags.scala
MetadataInputValidator

**Review by vimleshtna — COMMENTED** [review]
Thank you for doing this, good work. Just a couple of suggestions.

File: src/main/scala/uk/gov/nationalarchives/tdr/api/metadatainputvalidation/MetadataInputValidator.scala (hunk 3 above)
Comment (TomJKing) [reply src/main/scala/uk/gov/nationalarchives/tdr/api/metadatainputvalidation/MetadataInputValidator.scala:1]: I'd rather leave them in separate packages for now. Each package represents a different type check, eg authorisation, consignment state, metadata input. Its something we can reassess if we need to add/make further adjustments to the validators.

**Review by vimleshtna — APPROVED** [review]
Looks good, thanks

**Pull request merged**